code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm

```diff
@@ -0,0 +1,1195 @@
+			if(!item_image)
+				item_image = image(icon,null,"[item_type]_[our_state]")
+				item_image.color = item_color
+				item_image.appearance_flags = RESET_COLOR|KEEP_APART|PIXEL_SCALE
+				overlays_cache[item_key+our_state] = item_image
+			add_overlay(item_image)
+
+		if(teppi_wool)
+			var/image/wool_image = overlays_cache[wool_key+our_state+life_stage]	//Wool comes next, goes over top of the markings, is the same color too
+			if(!wool_image)
+				wool_image = image(icon,null,"wool_[our_state]")
+				wool_image.color = marking_color
+				wool_image.appearance_flags = RESET_COLOR|KEEP_APART|PIXEL_SCALE
+				overlays_cache[wool_key+our_state+life_stage] = wool_image
+			add_overlay(wool_image)
+
+	var/image/horn_image = overlays_cache[horn_key+our_state+life_stage+horn_type]		//Horns MUST come after marking and wool for layering purposes.
+	if(!horn_image)
+		if(!teppi_adult)
+			horn_image = image(icon,null,"horn_[our_state]")	//Babies only have one kind of horns
+		else
+			horn_image = image(icon,null,"horn_[our_state][horn_type]")
+		horn_image.color = horn_color
+		horn_image.appearance_flags = RESET_COLOR|KEEP_APART|PIXEL_SCALE
+		overlays_cache[horn_key+our_state+life_stage+horn_type] = horn_image
+	add_overlay(horn_image)
+
+	var/image/eye_image = overlays_cache[eye_key+our_state+life_stage]			//Eyes and skin should be above markings too, but their order doesn't matter
+	if(!eye_image)																//they won't intersect with eachother or the horns, but might intersect with some markings.
+		eye_image = image(icon,null,"eye_[our_state]")							//If we ever add horns or wool fluff that might cover them, remember to move these down as appropriate.
+		eye_image.color = eye_color												//Otherwise they will just always be on top of them.
+		eye_image.appearance_flags = RESET_COLOR|KEEP_APART|PIXEL_SCALE
+		overlays_cache[eye_key+our_state+life_stage] = eye_image
+	add_overlay(eye_image)
+
+	var/image/skin_image = overlays_cache[skin_key+our_state+life_stage]
+	if(!skin_image)
+		skin_image = image(icon,null,"skin_[our_state]")
+		skin_image.color = skin_color
+		skin_image.appearance_flags = RESET_COLOR|KEEP_APART|PIXEL_SCALE
+		overlays_cache[skin_key+our_state+life_stage] = skin_image
+	add_overlay(skin_image)
+	/////HIGHEST LAYER/////
+
+/mob/living/simple_mob/vore/alienanimals/teppi/attackby(var/obj/item/O as obj, var/mob/user as mob)
+	/////GRABS AND HOLDERS/////
+	if(istype(O, /obj/item/weapon/grab))
+		return ..()
+	if(istype(O, /obj/item/weapon/holder))
+		return ..()
+	if(user.a_intent != I_HELP) //be gentle
+		if(resting)
+			lay_down()
+		handle_affinity(user, -5)
+		user.visible_message(user, "<span class='notice'>\The [user] hits \the [src] with \the [O]. \The [src] grumbles at \the [user].</span>","<span class='notice'>You hits \the [src] with \the [O]. \The [src] grumbles at you.</span>")
+		playsound(src, 'sound/weapons/tap.ogg', 50, 1, -1)
+		return..()
```

Comment:
Pls change: Needs space, not even sure why this compiles. Byond may be more forgiving than I am